plugins/module_utils/module_helper.py

```diff
@@ -123,11 +125,11 @@ def wrapper(self, *args, **kwargs):
         except ModuleHelperException as e:
             if e.update_output:
                 self.update_output(e.update_output)
-            self.module.fail_json(changed=False, msg=e.msg, exception=traceback.format_exc(), output=self.output, vars=self.vars)
+            self.module.fail_json(msg=e.msg, exception=traceback.format_exc(), **self.output)
         except Exception as e:
-            self.vars.msg = "Module failed with exception: {0}".format(str(e).strip())
-            self.vars.exception = traceback.format_exc()
-            self.module.fail_json(changed=False, msg=self.vars.msg, exception=self.vars.exception, output=self.output, vars=self.vars)
+            msg = "Module failed with exception: {0}".format(str(e).strip())
+            exception = traceback.format_exc()
+            self.module.fail_json(msg=msg, exception=exception, **self.output)
```

Comment:
Removing `output=self.output` in both branches is a breaking change. Dropping `vars` as well.

Reply:
Ok, I can roll that back in but will deprecate those over time - both contents are coming into `**self.output` anyway.

Reply:
You probably need to have precautions in case `self.output` contains fields called `msg`, `exception`, `output`, or `vars`, because then the call to `fail_json` will fail with a Python error.

Reply:
done